### PR TITLE
Execute animations in a multithreaded way.

### PR DESCRIPTION
**Pull request by Beherith**
The config int AnimationMT allows run-time configuration of this being on or off.

Tested for sync safety. in #1139

Clean implementation of:
https://github.com/beyond-all-reason/spring/pull/1139

### CODE REVIEW

**Comment by lhog** [comment]
On my PC the MT execution is detrimental.

See: `CUnitScriptEngine::Tick`

ST
screen_2024-02-17_15-31-30-652

MT
screen_2024-02-17_15-31-20-529

Overall the difference is in the fraction of percentage. I'm not sure it's worth the effort.

**Comment by sprunk** [comment]
What was the test case? Perhaps try something like 3000 spinning windgens (happens in lategame BAR and ZK)

**Comment by lhog** [comment]
It was a standard BAR's `/luarules fightertest`, but yeah I can try with windfarms.

**Comment by lhog** [comment]
This spawns tanks by default, though you're right it's not the most animated unit in the world.

**Comment by lhog** [comment]
There was a bug in my updated, code, the MT version is faster.

File: rts/Sim/Units/Scripts/UnitScript.cpp [hunk 1]
```diff
@@ -218,7 +221,52 @@ bool CUnitScript::Tick(int deltaTime)
...
+			  new piece positions according to the animations
+*/
+void CUnitScript::TickAllAnims(int deltaTime)
+{
+	ZoneScoped;
+	// vector of indexes of finished animations,
+	// so we can get rid of them in constant time is stored in each units CUnitScript class at doneAnimsMT
+	// AnimContainerType doneAnimsMT[AMove + 1];
+	static constexpr TickAnimFunc tickAnimFuncs[AMove + 1] = { &CUnitScript::TickTurnAnim, &CUnitScript::TickSpinAnim, &CUnitScript::TickMoveAnim };
+	{
```
Comment (sprunk) [inline rts/Sim/Units/Scripts/UnitScript.cpp:235]: why the extra scope nesting?

Comment (lhog) [reply rts/Sim/Units/Scripts/UnitScript.cpp]: Fixed thanks

**Review by lhog — APPROVED** [review]

**Pull request merged**